packages/utils/src/field-type.ts

```diff
@@ -0,0 +1,415 @@
+                    value: fieldValue,
+                    enumerable: true,
+                    writable: false
+                });
+            }
+        }
+
+        return createObjectValue(result, true);
+    };
+}
+
+function formatTupleChildFields(childConfigs?: DataTypeFields) {
+    if (!childConfigs) {
+        return [];
+    }
+
+    const childFields: TFN[] = Object.entries(childConfigs)
+        .map(([field, config]) => {
+            const childConfig = getChildDataTypeConfig(
+                childConfigs!, field, config.type as FieldType
+            );
+            return coerceToType(config, childConfig);
+        });
+
+    return childFields;
+}
+
+function coerceToTuple(_fieldConfig: DataTypeFieldConfig, childConfig?: DataTypeFields) {
+    const childFields = formatTupleChildFields(childConfig);
+
+    return function _coerceToTuple(input: unknown) {
+        if (!isArrayLike(input)) {
+            throw new TypeError(`Expected ${toString(input)} (${getTypeOf(input)}) to be an array`);
+        }
+        const len = childFields.length;
+        if (input.length > len) {
+            throw new TypeError(`Expected ${toString(input)} (${getTypeOf(input)}) to have a length of ${len}`);
+        }
+
+        return createArrayValue(childFields.map((transformer, index) => {
+            const value = input[index];
+            return value != null ? transformer(value) : null;
+        }));
+    };
+}
+
+/** This is a low level api, only coerceToType should reference this,
+ * all other transforms should reference coerceToType as it handles arrays
+*/
+function getTransformerForFieldType<T = unknown>(
+    argFieldType: DataTypeFieldConfig,
+    childConfig?: DataTypeFields
+): (input: unknown) => T|unknown {
+    switch (argFieldType.type) {
+        case FieldType.String:
+        case FieldType.Text:
+        case FieldType.Keyword:
+        case FieldType.KeywordCaseInsensitive:
+        case FieldType.KeywordTokens:
+        case FieldType.KeywordTokensCaseInsensitive:
+        case FieldType.KeywordPathAnalyzer:
+        case FieldType.Domain:
+        case FieldType.Hostname:
+        case FieldType.NgramTokens:
+            return primitiveToString;
+        case FieldType.IP:
+            return isIPOrThrow;
+        case FieldType.IPRange:
+            return isIPRangeOrThrow;
+        case FieldType.Date:
+            return toEpochMSOrThrow;
+        case FieldType.Boolean:
+            return toBooleanOrThrow;
+        case FieldType.Float:
+            return toFloatOrThrow;
+        case FieldType.Number:
+        case FieldType.Double:
+            return toNumberOrThrow;
+        case FieldType.Byte:
+            return coerceToNumberType(FieldType.Byte);
+        case FieldType.Short:
+            return coerceToNumberType(FieldType.Short);
+        case FieldType.Integer:
+            return coerceToNumberType(FieldType.Integer);
+        case FieldType.Long:
+            return toBigIntOrThrow;
```

Comment:
It seems to me like `coerceToNumberType` should either be renamed to `coerceIntergerType` or be changed to handle all number types.

```suggestion
        case FieldType.Float:
        case FieldType.Number:
        case FieldType.Double:
        case FieldType.Byte:
        case FieldType.Short:
        case FieldType.Integer:
        case FieldType.Long:
            return coerceToNumberType(argFieldType.type);
```